plugin/cache/cache.go

```diff
@@ -36,6 +36,9 @@ type Cache struct {
 	duration   time.Duration
 	percentage int
 
+	serveStale bool
+	staleUpTo  time.Duration
```

Comment:
these two values can be unified into 1: if staleUpTo is zero it is disabled, otherwise we use it.

Reply:
How would you know when to serve immediately + fetch vs the normal behavior when the options is not present?

Reply:
isn't `serveStale == staleUpTo.IsZero()` ?

Reply:
If a configuration does not include serve_stale, we would have enabled the feature for ttl == 0.

Reply:
It breaks a prefetch test that expects to trigger a fetch after 30s for a TTL=30.

Reply:
Done.
I got waaaaaay derailed by one of my tests but in the end you were right and it was trivial to remove it. Thanks.